crates/astria-sequencer-relayer/src/relayer/submission.rs

```diff
@@ -0,0 +1,458 @@
+                },
+                post,
+            ) => {
+                ensure_consistent(sequencer_height, last_submission, post)
+                    .wrap_err("on-disk states are inconsistent")?;
+                Self {
+                    pre,
+                    post,
+                    pre_path,
+                    post_path,
+                }
+            }
+        };
+        Ok(state)
+    }
+}
+
+fn ensure_consistent(
+    sequencer_height_started: u64,
+    last_submission: PostSubmission,
+    current_submission: PostSubmission,
+) -> eyre::Result<()> {
+    ensure_last_and_current_are_different(last_submission, current_submission)?;
+    ensure_last_is_not_submitted_while_current_is_fresh(last_submission, current_submission)?;
+    ensure_height_in_last_does_not_exceed_height_in_current(last_submission, current_submission)?;
+    ensure_next_height_is_not_less_than_current(sequencer_height_started, current_submission)?;
+    Ok(())
+}
+
+fn ensure_last_and_current_are_different(
+    last_submission: PostSubmission,
+    current_submission: PostSubmission,
+) -> eyre::Result<()> {
+    ensure!(
+        last_submission != current_submission,
+        "the last post-submission state recorded in the pre-submit file matches the state in the \
+         post-submission file. This indicates that either submission to Celestia failed, or that \
+         relayer failed to write its post-submission state to disk. Verify that the sequencer \
+         height in the pre-submit file was written to Celestia and update the post-submission \
+         file with the sequencer and Celestia heights or, if you want to have relayer start \
+         submitting from the currently set post-submission file, update the pre-submission file \
+         to read `{{\"state\": \"ignore\"}}`"
+    );
+    Ok(())
+}
+fn ensure_last_is_not_submitted_while_current_is_fresh(
+    last_submission: PostSubmission,
+    current_submission: PostSubmission,
+) -> eyre::Result<()> {
+    ensure!(
+        !(last_submission.is_submitted() && current_submission.is_fresh()),
+        "the submission recorded in the post-submit file cannot be `fresh` while \
+         `last_submission` in the pre-submit file is `submitted`",
+    );
+    Ok(())
+}
+
+fn ensure_height_in_last_does_not_exceed_height_in_current(
+    last_submission: PostSubmission,
+    current_submission: PostSubmission,
+) -> eyre::Result<()> {
+    let PostSubmission::Submitted {
+        sequencer_height: height_in_last,
+        ..
+    } = last_submission
+    else {
+        return Ok(());
+    };
+    let PostSubmission::Submitted {
+        sequencer_height: height_in_current,
+        ..
+    } = current_submission
+    else {
+        return Ok(());
+    };
+    ensure!(
+        height_in_last <= height_in_current,
+        "sequencer height in `last_submission` in the pre-submit file cannot exceed the sequencer \
+         height recorded as submitted in the post-submit file"
```

Comment:
```suggestion
        "sequencer height in `last_submission` in the pre-submit file cannot exceed or be equal to the sequencer \
         height recorded as submitted in the post-submit file"
```

Reply:
I have tightened these tests (similar reason to the one below): `ensure_height_in_last_is_less_than_height_in_current` now enforces that the newest successful submission is larger than the previous successful submission.